packages/python/plotly/plotly/express/_core.py

```diff
@@ -1295,11 +1295,17 @@ def build_dataframe(args, constructor):
     # make copies of all the fields via dict() and list()
     for field in args:
         if field in array_attrables and args[field] is not None:
-            args[field] = (
-                dict(args[field])
-                if isinstance(args[field], dict)
-                else list(args[field])
-            )
+            if isinstance(args[field], dict):
+                args[field] = dict(args[field])
+            elif (
+                field in ["custom_data", "hover_data"]
+                and isinstance(args[field], str)
+                and "data_frame" in args.keys()
```

Comment:
here I wouldn't check if the string is in the columns... there is other machinery elsewhere that does this kind of check and has better error messages etc, and simply checking here will disable this I think.

Reply:
1. I took out the check for the string is in the columns which I agree was a good call. Now if I pass a single string that's not a column name I get the better error message `ValueError: Value of 'hover_data_0' is not the name of a column in 'data_frame'...`. 
2. All tests are now passing in `test_px` and I've added one new test and modified an existing one to account for both strings and dicts (and lists which wasn't present before).
    a. _Across all of test_optional I get `766 passed, 16 skipped, 79 warnings`, not too sure why some get skipped._
    b. _In test_core I fail 8 tests but all were `UnicodeDecodeError: 'charmap' codec can't decode byte 0x8f in position 291695` which I presume to be unrelated_
    c. I also needed to install various packages that weren't in the [optional-requirements.txt](https://github.com/plotly/plotly.py/blob/master/packages/python/plotly/optional-requirements.txt) such as `statsmodels`, `scikit-image`, and `kaleido`. I believe there's others but I decided to focus on just test_optional and test_core. I may submit this as an additional issue and PR if that would be desirable!

![image](https://user-images.githubusercontent.com/22160528/223000645-3467ab08-e9c5-48eb-9371-d370219a605b.png)

Reply:
The tests look good to me on CI, thanks! To run the tests you need to install one of the testing requirements files from here https://github.com/plotly/plotly.py/tree/master/packages/python/plotly/test_requirements (which should probably be documented in the contributing guide!!)

Reply:
Ah thanks! It looks like the [contributing guide](https://github.com/plotly/plotly.py/blob/master/contributing.md) (assuming that's the right one to look at) only mentions optional-requirements. 
 
![image](https://user-images.githubusercontent.com/22160528/223002950-fc08858f-d988-4728-9e1f-9b9d3cb43dfe.png)